SourceCode/h8/light.cpp

```diff
@@ -0,0 +1,796 @@
+            assert(!positionValue.IsNull() && positionValue.IsArray());
+            cameraPosition = loadVector(positionValue.GetArray());
+        }
+
+        const rapidjson::Value& ligthsValue = document.FindMember("lights")->value;
+        if(!ligthsValue.IsNull() && ligthsValue.IsArray()) {
+            for(rapidjson::Value::ConstValueIterator itr = ligthsValue.Begin(); 
+                    itr != ligthsValue.End(); 
+                    itr++)
+            {
+                const rapidjson::Value& intensityValue 
+                        = itr->FindMember("intensity")->value;
+                assert(!intensityValue.IsNull() && intensityValue.IsInt());
+
+                const rapidjson::Value& positionValue 
+                        = itr->FindMember("position")->value;
+                assert(!positionValue.IsNull() && positionValue.IsArray());
+                lights.emplace_back(
+                    loadVector(positionValue.GetArray()), 
+                    intensityValue.GetInt()
+                );
+            }
+        }
+
+        const rapidjson::Value& objectsValue = document.FindMember("objects")->value;
+        if(!objectsValue.IsNull() && objectsValue.IsArray()) {
+            for(rapidjson::Value::ConstValueIterator itr = objectsValue.Begin(); 
+                    itr != objectsValue.End(); 
+                    itr++)
+            {
+                const rapidjson::Value& verticesValue 
+                            = itr->FindMember("vertices")->value;
+                assert(!verticesValue.IsNull() && verticesValue.IsArray());
+                loadVertices(verticesValue.GetArray());
+
+                const rapidjson::Value& trianglesValue 
+                            = itr->FindMember("triangles")->value;
+                assert(!trianglesValue.IsNull() && trianglesValue.IsArray());
+                loadTriangles(trianglesValue.GetArray());
+            }
+        }
+    }
+
+    std::string getFileName() const {
+        return fileName;
+    }
+    unsigned int getImageWidth() const {
+        return settings.imageSettings.width;
+    }
+    unsigned int getImageHeight() const {
+        return settings.imageSettings.height;
+    }
+    Vector getBgColor() const {
+        return settings.bgColor;
+    }
+    std::vector<Mesh> getMeshes() const {
+        return meshes;
+    }
+    Vector getCameraPosition() const {
+        return cameraPosition;
+    }
+    Matrix getInitialTransformation() const {
+        return initialTransformation;
+    }
+    std::vector<Light> getLights() const {
+        return lights;
+    }
+
+    void setFileName(const std::string fileName) {
+        this->fileName = fileName;
+    }
+};
+
+class Renderer {
+    struct {
+        Vector bgColor;
+        struct {
+            unsigned int width;
+            unsigned int height;
+            unsigned int maxColorComponent;
+        } imageSettings;
+    } settings;
+    std::vector<Mesh> meshes;
+    Vector cameraPosition;
```

Comment:
You could use separate camera class

Reply:
I have a separate class which I use. In this case I use a vector to save space and speed, because the input file gives only the position. @VladislavVulchevChaos